tests/lib/screenshot-testing/support/chai-extras.js

```diff
@@ -501,14 +501,37 @@ function performAutomaticPageChecks(done) {
 
 function checkForDangerousLinks() {
     var links = pageRenderer.webpage.evaluate(function () {
-        var result = [];
-        $('a').each(function () {
-            var href = $(this).attr('href');
-            if (/^(javascript|vbscript|data):;*[^;]+/.test(href) && !(/^javascript:void\(0\);?$/.test(href))) {
-                result.push($(this).text() + ' - [href = ' + href + ']');
+        try {
+            var result = [];
+
+            var safeJsLinks = [
+                /^javascript:void\(0\);?$/,
+                /^javascript:window\.history\.back\(\);?$/,
+                /^javascript:window\.location\.reload\(\);?$/
+            ];
+
+            var linkElements = document.getElementsByTagName('a');
+
+            elements:
+            for (var i = 0; i !== linkElements.length; ++i) {
+                var element = linkElements.item(i);
+
+                var href = element.getAttribute('href');
+                if (/^(javascript|vbscript|data):;*[^;]+/.test(href)) {
+                    for (var index in safeJsLinks) {
+                        if (safeJsLinks[index].test(href)) {
```

Comment:
Is there any reason why you keep using regex here? Wouldn't it be easier to just test via `Array.prototype.indexOf()`? You are using `^` and `$` so, if you do not plan to use wildcards in safe links any time soon, it would be performanter to use index of.